oauth/login_utils.py

```diff
@@ -93,7 +96,17 @@ def get_sub_username_email_from_token(
                 config.get("PREFERRED_GROUP_CLAIM_NAME", None)
             )
 
-    return decoded_id_token["sub"], lusername, email_address, additional_info
+    # Call the login service's get_user_id() if it exists (e.g., RHSSO), otherwise use sub directly
+    if (
```

Comment:
instead of this logic, can you make `get_user_id` a required function in the base metaclass and have the subclass use the generic function of returning the `sub` in the token

Reply:
moved `get_user_id()` to `OAuthService` base meta class and added the override for rhsso